packages/components/src/internal/components/labels/APIWrapper.ts

```diff
@@ -131,6 +178,33 @@ export class LabelPrintingServerAPIWrapper implements LabelPrintingAPIWrapper {
             });
         });
     };
+
+    ensureLabelTemplatesList = (): Promise<LabelTemplate[]> => {
```

Comment:
My thought was to do it here because the only time the list would get deleted would be after folder creation, so if it was done only then, it isn't regenerated. 

The permissions issue is interesting, my general thought is that if the list doesn't exist and the user can't create it then:
1) they can't get to the admin page anyway
1) they can't print because there are no configured labels to print -- (the options should to export/print should be hidden)

Added a check for AppAdmin permissions prior to attempting to recreate list.